src/rules/selector-class-pattern/index.js

```diff
@@ -13,23 +14,46 @@ export const messages = ruleMessages(ruleName, {
   expected: selectorValue => `Expected class selector ".${selectorValue}" to match specified pattern`,
 })
 
-export default function (pattern) {
+export default function (pattern, options) {
   return (root, result) => {
     const validOptions = validateOptions(result, ruleName, {
       actual: pattern,
       possible: [ isRegExp, isString ],
+    }, {
+      actual: options,
+      possible: {
+        resolveNestedSelectors: isBoolean,
+      },
+      optional: true,
     })
     if (!validOptions) { return }
 
+    // In context of a nested selector, some combinations
+    // are tested mutliple times, to avoid triggering multiple
+    // identical errors, we will use an internal list of checked rules
+    var checked = []
```

Comment:
Don't we *want* to trigger multiple identical errors if identical selectors are used multiple times?

For example, if my pattern is `/[a-z]+/` and I have this code:

```css
.foo {}
.foo .Bar {}
.foo {
  & .Bar {}
}
```

Don't I want *two* warnings, one for each `.foo .Bar`? 

(Or am I misunderstanding this point?)